warehouse/static/js/warehouse/controllers/password_controller.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["showPassword", "password"];
+
+  connect() {
+    if (this.passwordTargets && this.showPasswordTarget) {
+      // Reset these so they don't persist between page reloads
+      for (let field of this.passwordTargets) {
+        field.type = "password";
+        this.showPasswordTarget.checked = false;
+      }
+    }
```

Comment:
This function is only going to get called if we've set the `data-controller="password"` attribute on some field, so we probably don't need the checks anymore as this will only run on pages that we already know have the targets. It's also unnecessarily setting `this.showPasswordTarget.checked`  for every field instead of just once (but this was an issue from the original implementation). I think this can just become something like:

```js
connect() {
    // Reset the targets so they don't persist between page reloads
    this.showPasswordTarget.checked = false;
    this.togglePasswords();
}
```